pkgs/build-support/dotnet/build-dotnet-module/default.nix

```diff
@@ -41,6 +41,7 @@
   # Unfortunately, dotnet has no method for doing this automatically.
   # If unset, all executables in the projects root will get installed. This may cause bloat!
 , executables ? null
+, dontPublish ? false
```

Comment:
`packNupkg` isn't used in nixpkgs outside of the test project, so I think it would be safe to disable publish when packing.

However, I imagined long-term that there'd be a separate outputs for loose builds and nupkgs.